src/test/java/it/eng/idsa/businesslogic/service/impl/ProtocolValidationServiceImplTest.java

```diff
@@ -0,0 +1,129 @@
+package it.eng.idsa.businesslogic.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import it.eng.idsa.businesslogic.service.RejectionMessageService;
+import it.eng.idsa.businesslogic.util.RejectionMessageType;
+
+public class ProtocolValidationServiceImplTest {
+	
+    private static final String HTTPS = "https";
+	
+	private static final String HTTP = "http";
+	
+	private static final String WSS = "wss";
+	
+	@InjectMocks
+	private ProtocolValidationServiceImpl protocolValidationServiceImpl;
+	
+	private boolean websocketBetweenECC;
+	
+	private boolean jetty;
+	
+	private String requiredECCProtocol;
+	
+	@Mock
+	private RejectionMessageService rejectionMessageService;
+
+	@BeforeEach
+	public void setup() {
+		MockitoAnnotations.initMocks(this);
+		requiredECCProtocol = setRequiredProtocol(HTTPS);
+		mockFieldsToProtocol(HTTPS);
+	}
+	
+	
+	@Test
+	public void validateProtocolNoDelimiterInForwardTo() {
+		String forwardTo = "mock.com";
+		ReflectionTestUtils.setField(protocolValidationServiceImpl, "validateProtocol", true);
+		protocolValidationServiceImpl.validateProtocol(forwardTo);
+		verify(rejectionMessageService).sendRejectionMessage(RejectionMessageType.REJECTION_MESSAGE_LOCAL_ISSUES, null);
+	}
+	
+	@Test
+	public void validateProtocolWrongProtocolInForwardTo() {
+		String forwardTo = "http://mock.com";
+		ReflectionTestUtils.setField(protocolValidationServiceImpl, "validateProtocol", true);
+		protocolValidationServiceImpl.validateProtocol(forwardTo);
+		verify(rejectionMessageService).sendRejectionMessage(RejectionMessageType.REJECTION_MESSAGE_LOCAL_ISSUES, null);
+	}
+	
+	@Test
+	public void validateProtocolCorrectForwardTo() {
+		String forwardTo = "https://mock.com";
+		ReflectionTestUtils.setField(protocolValidationServiceImpl, "validateProtocol", true);
+		String testForwardTo = protocolValidationServiceImpl.validateProtocol(forwardTo);
+		verify(rejectionMessageService, times(0)).sendRejectionMessage(RejectionMessageType.REJECTION_MESSAGE_LOCAL_ISSUES, null);
+		assertEquals(forwardTo, testForwardTo);
+	}
+	
```

Comment:
I was thinking whether we needed it since it's just copy-pasting the same thing. And if one works, all work, because it's not depending one those combinations. I can add them if you think that they are necessary?

Reply:
Please add them, so that we are covered.